src/ResourceManager/ServiceBus/Commands.ServiceBus/Cmdlets/Queue/RemoveAzureServiceBusQueue.cs

```diff
@@ -14,32 +14,62 @@
 
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
 using System.Management.Automation;
+using Microsoft.Azure.Commands.ServiceBus.Models;
 namespace Microsoft.Azure.Commands.ServiceBus.Commands.Queue
 {
     /// <summary>
     /// 'Remove-AzureRmServiceBusQueue' Cmdlet removes the specified Queue
     /// </summary>
-    [Cmdlet(VerbsCommon.Remove, ServicebusQueueVerb, SupportsShouldProcess = true), OutputType(typeof(bool))]
+    [Cmdlet(VerbsCommon.Remove, ServicebusQueueVerb, DefaultParameterSetName = QueuePropertiesParameterSet, SupportsShouldProcess = true), OutputType(typeof(bool))]
     public class RemoveAzureRmServiceBusQueue : AzureServiceBusCmdletBase
     {
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "The name of the resource group")]
+        [Parameter(Mandatory = true, ParameterSetName = QueuePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "The name of the resource group")]
         [ResourceGroupCompleter]
         [Alias("ResourceGroup")]
         [ValidateNotNullOrEmpty]
         public string ResourceGroupName { get; set; }
 
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [Parameter(Mandatory = true, ParameterSetName = QueuePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
         [Alias(AliasNamespaceName)]
         [ValidateNotNullOrEmpty]
         public string Namespace { get; set; }
 
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "Queue Name")]
+        [Parameter(Mandatory = true, ParameterSetName = QueuePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "Queue Name")]
         [Alias(AliasQueueName)]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
+        [Parameter(Mandatory = true, ParameterSetName = QueueInputObjectParameterSet, ValueFromPipeline = true, Position = 0, HelpMessage = "Service Bus Queue Object")]
+        [ValidateNotNullOrEmpty]
+        public PSQueueAttributes InputObject { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = QueueResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Service Bus Queue Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
```

Comment:
@v-Ajnava it doesn't look like the `-PassThru` parameter is being used anymore. I might have been a little unclear with the comment I made regarding this. I wanted to make sure that whatever was returned by the operation below was still being returned. You can still assign the result to a variable, and if the user provides `-PassThru`, then you write that object to the output stream.

```cs
var result = Client.DeleteQueue(ResourceGroupName, Namespace, Name);
if (PassThru.IsPresent)
{
    WriteObject(result);
}
```

Reply:
@v-Ajnava this comment applies to the other affected areas